### PR TITLE
prepare 1.8 release

### PR DESCRIPTION
**Pull request by fscherf**
This PR contains some small new features, bugfixes and removes some previously deprecated features

### CODE REVIEW

File: lona/view.py [hunk 1]
```diff
@@ -45,24 +38,6 @@ def __init__(
         self._view_runtime: ViewRuntime = view_runtime
         self._request: Request = request
 
-    # objects #################################################################
-    @classmethod
-    def iter_objects(cls: type[V]) -> Iterator[V]:
-        # TODO: remove after 1.8
-
-        warnings.warn(
-            'LonaView.iter_objects() will be removed in 1.8',
```
Comment (maratori) [inline lona/view.py:54]: Don't we need to remove this from docs as well?

File: lona/view_loader.py [hunk 2]
```diff
@@ -66,15 +64,6 @@ def _run_checks(self, route: Route, view: type[LonaView]) -> None:
                     hook_name,
                 )
 
-        # TODO: remove after 1.8
-        if(isinstance(view, type) and
-           view.on_shutdown is not LonaView.on_shutdown):
-
-            warnings.warn(
-                'LonaView.on_shutdown() will be removed in 1.8',
```
Comment (maratori) [inline lona/view_loader.py:74]: The same as above

File: lona/view.py (hunk 1 above)
Comment (fscherf) [reply lona/view.py:54]: I will leave the docs for this method in the docs until 2.0, but i will add ``note: removed in 1.8`` like before. I do this usually when writing the changelog.

File: lona/view_loader.py (hunk 2 above)
Comment (fscherf) [reply lona/view_loader.py:74]: The same as above :)

**Review by maratori — APPROVED** [review]
LGTM

**Comment by laundmo** [comment]
LGTM, haven't tested but the code seems clean

**Review by laundmo — APPROVED** [review]

**Comment by fscherf** [comment]
I found one last line using ``LonaView.on_shutdown`` while testing and fixed it.

@maratori, @laundmo: Thanks for reviewing!

**Comment by mwx23** [comment]
Woohoo new `lona` version! I'm excited! :)

**Pull request merged**